var/spack/repos/builtin/packages/py-fenics-ffcx/package.py

```diff
@@ -36,19 +37,23 @@ class PyFenicsFfcx(PythonPackage):
     )
 
     depends_on("python@3.7:", type=("build", "run"))
-    depends_on("py-setuptools@58:", type="build", when="@0.4:")
-
-    depends_on("py-cffi", type="run")
-    depends_on("py-numpy", type="run")
-
-    depends_on("py-fenics-ufl@main", type="run", when="@main")
-    depends_on("py-fenics-ufl@2022.2.0", type="run", when="@0.5.0:0.5")
-    depends_on("py-fenics-ufl@2022.1.0", type="run", when="@0.4.2")
-    depends_on("py-fenics-ufl@2021.1.0", type="run", when="@0.1.0:0.3")
-
-    depends_on("py-fenics-basix@main", type="run", when="@main")
-    depends_on("py-fenics-basix@0.5.1:0.5", type="run", when="@0.5.0:0.5")
-    depends_on("py-fenics-basix@0.4.2", type="run", when="@0.4.2")
-    depends_on("py-fenics-basix@0.3.0", type="run", when="@0.3.0")
-    depends_on("py-fenics-basix@0.2.0", type="run", when="@0.2.0")
-    depends_on("py-fenics-basix@0.1.0", type="run", when="@0.1.0")
+    depends_on("py-setuptools@58:", type=("build", "run"), when="@0.4:")
+    depends_on("py-setuptools@40:", type=("build", "run"))
+
+    depends_on("py-cffi", type=("build", "run"))
+    depends_on("py-numpy@1.21:", type=("build", "run"), when="@0.4:")
+    depends_on("py-numpy@1.15:", type=("build", "run"))
```

Comment:
Is the numpy min version documented anywhere?

Reply:
My mistake. 1.21 is the min version. Versions lower do not work with py3.7+